### PR TITLE
Fix logging on the simulator

### PR DESCRIPTION
**Pull request by iFrostizz**
Closes https://github.com/ava-labs/hypersdk/issues/864, closes https://github.com/ava-labs/hypersdk/issues/850, closes https://github.com/ava-labs/hypersdk/issues/787
The logging on the simulator was broken. The issue is that the logging level which is passed to subcommands was done before the call to `ExecuteContext`. We call `Execute` and it's a bit hacky but there doesn't seem to be many very clean solutions. The logging can now be disabled using the new `--log-disable-display-plugin-logs` flag.

### CODE REVIEW

File: x/programs/cmd/simulator/cmd/logger.go [hunk 1]
```diff
@@ -48,17 +48,17 @@ func (f *logFactory) makeLogger(config logging.Config) (logging.Logger, error) {
...
-	consoleEnc := config.LogFormat.ConsoleEncoder()
+	consoleEnc := logging.Colors.ConsoleEncoder()
 	fileEnc := config.LogFormat.FileEncoder()
 
 	var consoleWriter io.WriteCloser
 	if config.DisableWriterDisplaying {
 		consoleWriter = newDiscardWriteCloser()
 	} else {
-		consoleWriter = os.Stdout
+		consoleWriter = os.Stderr
```
Comment (iFrostizz) [inline x/programs/cmd/simulator/cmd/logger.go:58]: We use Stderr in order not to interfere with the plans being piped from Stdin. This is hopefully a temporary solution.

File: x/programs/cmd/simulator/cmd/root.go [hunk 2]
```diff
@@ -48,21 +57,33 @@ func NewRootCmd() *cobra.Command {
...
 	cmd.SilenceErrors = true
 	cmd.SetHelpCommand(&cobra.Command{Hidden: true})
-	cmd.PersistentFlags().StringVar(&s.logLevel, "log-level", "info", "log level")
+
+	// pre-execute the command to pre-parse flags
+	err := cmd.Execute()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
```
[the remark this answers is not in the document]
Comment (najeal) [reply x/programs/cmd/simulator/cmd/root.go:83]: I would suggest to make a `cmd.ParseFlags(os.Args[1:])` instead. This is cleaner in my opinion. What do you think?

**Review by najeal — COMMENTED** [review]
Just proposing another solution for accessing the value of the `log-level` flag

File: x/programs/cmd/simulator/cmd/root.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (iFrostizz) [reply x/programs/cmd/simulator/cmd/root.go:83]: @najeal the issue is that this won't let us place the cli flag in another position than just after the call to the binary if I understand correctly

[the remark this answers is not in the document]
Comment (iFrostizz) [reply x/programs/cmd/simulator/cmd/root.go:83]: I have read too fast, that's a good suggestion and I will add it thanks !

**Pull request closed without merging**